src/ui.cpp

```diff
@@ -585,304 +580,75 @@ void uilist::calc_data()
         } while( !assigned && hotkey != input_event() );
     }
 
-    if( desc_enabled ) {
-        if( descwidth_final > TERMX ) {
-            desc_enabled = false; // give up
-        } else if( descwidth_final > w_width ) {
-            w_width = descwidth_final;
-        }
-
-    }
+    vmax = entries.size();
+    uint additional_lines = 0;
 
-    if( !text.empty() ) {
-        int twidth = utf8_width( remove_color_tags( text ) );
-        bool formattxt = true;
-        int realtextwidth = 0;
-        if( textwidth == -1 ) {
-            if( !w_auto ) {
-                realtextwidth = w_width - 4;
-            } else {
-                realtextwidth = twidth;
-                if( twidth + 4 > w_width ) {
-                    if( realtextwidth + 4 > TERMX ) {
-                        realtextwidth = TERMX - 4;
-                    }
-                    textformatted = foldstring( text, realtextwidth );
-                    formattxt = false;
-                    realtextwidth = 10;
-                    for( auto &l : textformatted ) {
-                        const int w = utf8_width( remove_color_tags( l ) );
-                        if( w > realtextwidth ) {
-                            realtextwidth = w;
-                        }
-                    }
-                    if( realtextwidth + 4 > w_width ) {
-                        w_width = realtextwidth + 4;
-                    }
-                }
-            }
-        } else if( textwidth != -1 ) {
-            realtextwidth = textwidth;
-            if( realtextwidth + 4 > w_width ) {
-                w_width = realtextwidth + 4;
-            }
-        }
-        if( formattxt ) {
-            textformatted = foldstring( text, realtextwidth );
-        }
+    bool has_titlebar = title[0] != '#';
+    if( has_titlebar ) {
+        additional_lines += 1;
     }
 
-    // shrink-to-fit
+    auto desc_lines = 0;
```

Comment:
I am surprised it didn't warn about this auto too.
```suggestion
    int desc_lines = 0;
```